frodo.js

```diff
@@ -8,125 +8,127 @@
  *
```

Comment:
SCRIPT-Well done with the efficient error handling, it covers all the possible scenarios.